homeassistant/components/asuswrt/bridge.py

```diff
@@ -47,6 +59,42 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def migrate_legacy_protocols(
+    hass: HomeAssistant, conf: dict[str, Any]
+) -> dict[str, Any] | None:
+    """Migrate router protocol from legacy to http."""
+    if conf.get(CONF_SSH_KEY):
+        # migration temptative is not possible with ssh key
+        return None
+
+    if is_http(conf[CONF_PROTOCOL]):
+        return {}
+
+    sec_conf = {
+        k: v for k, v in conf.items() if k in [CONF_HOST, CONF_USERNAME, CONF_PASSWORD]
+    }
+    for protocol in HTTP_PROTOCOLS:
```

Comment:
This is a security risk, as one can unexpectedly make someone go and use insecure `http` (e.g., MITM attack, blocking HTTPS).

Reply:
What doo you exactly suggest? Asus Router normally have HTTP protocol enabled but not HTTPS. We should only try with HTTPS? In most of case probably it will fail.